File: python/medousa-sdk/src/medousa/mcp_gateway.py
from __future__ import annotations

from medousa._decode import decode
from medousa._ops import op_path
from medousa.client import MedousaClient
from medousa.types import McpGatewayStatusResponse


class McpGatewayApi:
    def __init__(self, client: MedousaClient) -> None:
        self._client = client

    async def status(self) -> McpGatewayStatusResponse:
        value = await self._client.transport.get_json(
            self._client.base_url,
            op_path("mcp.gateway.status.get"),
        )
        return decode(McpGatewayStatusResponse, value)

File: crates/medousa-sdk/src/mcp_gateway.rs
#[cfg(feature = "async")]
use medousa_types::McpGatewayStatusResponse;

#[cfg(feature = "async")]
use crate::client::MedousaClient;
use crate::generated::ops;
#[cfg(feature = "async")]
use crate::transport::decode;

#[cfg(feature = "async")]
pub struct McpGatewayApi<'a> {
    pub(crate) client: &'a MedousaClient,
}

#[cfg(feature = "async")]
impl McpGatewayApi<'_> {
    pub async fn status(&self) -> Result<McpGatewayStatusResponse, crate::SdkError> {
        let value = self
            .client
            .transport()
            .get_json(self.client.base_url(), ops::MCP_GATEWAY_STATUS_GET.path)
            .await?;
        decode(value).await
    }
}
